mozci/data/sources/treeherder/__init__.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+
+from loguru import logger
+
+from mozci.data.base import DataSource
+
+try:
+    from treeherder.model.models import Job
```

Comment:
You should add this as an optional dependency, e.g:
```
$ poetry add --optional treeherder
```
This will update the pyproject.toml and poetry.lock files automatically (just commit whatever they are after that).

Reply:
Actually, looks like this package might not be on PYPI. Will this import only work on your machine? Would be nice to have more general support.

Maybe if you don't want to add this to pypi, we could add a configuration option that takes a path to the module which gets added to `sys.path` here?

Reply:
Yeah, Treeherder is not on PYPI.  This will only work when `mozci` is installed on a Treeherder instance.  The import references `Job` which is a Django model.  I wasn't sure what the best way was to do a *conditional* import for this.

I figured this way, it would try to do the import, and just punt if it can't find it.  Is adding it to `sys.path` cleaner?